plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -443,7 +443,7 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
 
          /* de-dupe here... no point in aborting block if we already know the block */
          auto existing = chain.fetch_block_by_id( id );
-         if( existing ) { return false; }
+         if( existing ) { return true; } // return true because the block is valid
```

Comment:
It is actually not possible for this to return `true` when called from `net_plugin` since `net_plugin` checks this before calling `accept_block`. The return value is ignored in `chain_plugin::push_block` not that anyone has ever used `push_block` or ever will, but if they did and pushed a duplicate block without this check the application would shutdown() on the `fork_database_exception`. 

I changed it to return `true` to match what happens in `net_plugin` when it finds it exists before calling `accept_block`. Just seems cleaner.

Reply:
OK, thanks a lot for the explanation Kevin!